storage/bucket.go

```diff
@@ -2124,17 +2108,8 @@ func (it *ObjectIterator) fetch(pageSize int, pageToken string) (string, error)
 //
 // Note: The returned iterator is not safe for concurrent operations without explicit synchronization.
 func (c *Client) Buckets(ctx context.Context, projectID string) *BucketIterator {
-	it := &BucketIterator{
-		ctx:       ctx,
-		client:    c,
-		projectID: projectID,
-	}
-	it.pageInfo, it.nextFunc = iterator.NewPageInfo(
-		it.fetch,
-		func() int { return len(it.buckets) },
-		func() interface{} { b := it.buckets; it.buckets = nil; return b })
-
-	return it
+	o := makeStorageOpts(true, c.retry, "")
```

Comment:
Why `""` for userProject? Is that not in scope for this RPC?

Reply:
UserProject is a method on BucketHandle, so there's no way for library users to actually set UserProject on these top level calls: https://pkg.go.dev/cloud.google.com/go/storage#BucketHandle.UserProject

Maybe we should add a method on client as well but that's outside the scope here.

Reply:
Makes sense. Thanks!